src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssetsBaselineComparer.cs

```diff
@@ -0,0 +1,251 @@
+Start by ensuring you're dogfooding the SDK from the current branch (dotnet --version should be '*.0.0-dev').
+    If you're not on the dogfood sdk, from the root of the repository run:
+        1. dotnet clean
+        2. .\restore.cmd
+        3. .\build.cmd
+        4. .\eng\dogfood.cmd
+
+Then, using the dogfood SDK run the .\src\RazorSdk\update-test-baselines.ps1 script.";
+
+    public static StaticWebAssetsBaselineComparer Instance { get; } = new();
+
+    internal void AssertManifest(StaticWebAssetsManifest expected, StaticWebAssetsManifest manifest)
+    {
+        //Many of the properties in the manifest contain full paths, to avoid flakiness on the tests, we don't compare the full paths.
+        manifest.Version.Should().Be(expected.Version);
+        manifest.Source.Should().Be(expected.Source);
+        manifest.BasePath.Should().Be(expected.BasePath);
+        manifest.Mode.Should().Be(expected.Mode);
+        manifest.ManifestType.Should().Be(expected.ManifestType);
+
+        manifest.ReferencedProjectsConfiguration.Should().HaveSameCount(expected.ReferencedProjectsConfiguration);
+
+        // Relax the check for project reference configuration items see
+        // https://github.com/dotnet/sdk/pull/27381#issuecomment-1228764471
+        // for details.
+        //manifest.ReferencedProjectsConfiguration.OrderBy(cm => cm.Identity)
+        //    .Should()
+        //    .BeEquivalentTo(expected.ReferencedProjectsConfiguration.OrderBy(cm => cm.Identity));
+
+        manifest.DiscoveryPatterns.OrderBy(dp => dp.Name).Should().BeEquivalentTo(expected.DiscoveryPatterns.OrderBy(dp => dp.Name));
+
+        var manifestAssets = manifest.Assets
+            .OrderBy(a => a.BasePath)
+            .ThenBy(a => a.RelativePath)
+            .ThenBy(a => a.AssetKind)
+            .GroupBy(a => GetGroup(a))
+            .ToDictionary(a => a.Key, a => a.ToArray());
+
+        var expectedAssets = expected.Assets
+            .OrderBy(a => a.BasePath)
+            .ThenBy(a => a.RelativePath)
+            .ThenBy(a => a.AssetKind)
+            .GroupBy(a => GetGroup(a))
+            .ToDictionary(a => a.Key, a => a.ToArray());
+
+        foreach (var (group, manifestAssetsGroup) in manifestAssets)
+        {
+            var expectedAssetsGroup = expectedAssets[group];
+            CompareGroup(group, manifestAssetsGroup, expectedAssetsGroup);
+        }
+    }
+
+    protected virtual void CompareGroup(string group, StaticWebAsset[] manifestAssets, StaticWebAsset[] expectedAssets)
+    {
+        var comparisonMode = CompareAssetCounts(group, manifestAssets, expectedAssets);
+
+        // Otherwise, do a property level comparison of all assets
+        switch (comparisonMode)
+        {
+            case GroupComparisonMode.Exact:
+                break;
+            case GroupComparisonMode.AllowAdditionalAssets:
+                break;
+            default:
+                break;
+        }
```

Comment:
TODO: we plan to allow tests to pass even if the baselines do not exactly match in some circumstances (like if there are additional dlls and so on). 

For that, we will compare the list of assets by groups of assets instead of as a flat list and apply different validation rules to each group.